CMakeLists.txt

```diff
@@ -567,9 +575,13 @@ option(AVIF_ENABLE_GTEST
 )
 
 if(AVIF_BUILD_APPS OR (AVIF_BUILD_TESTS AND AVIF_ENABLE_GTEST))
-    find_package(ZLIB REQUIRED)
-    find_package(PNG 1.6.32 REQUIRED) # 1.6.32 or above for png_get_eXIf_1()/png_set_eXIf_1() and iTXt (for XMP).
-    find_package(JPEG REQUIRED)
+    if(NOT AVIF_LOCAL_ZLIBPNG)
+        find_package(ZLIB REQUIRED)
+        find_package(PNG 1.6.32 REQUIRED) # 1.6.32 or above for png_get_eXIf_1()/png_set_eXIf_1() and iTXt (for XMP).
+    endif()
+    if(NOT AVIF_LOCAL_JPEG)
+        find_package(JPEG REQUIRED)
+    endif()
```

Comment:
I tested the `AVIF_LOCAL_ZLIBPNG` and `AVIF_LOCAL_JPEG` changes here on my Linux desktop. These changes work, but I don't know why they work.

For example, suppose `AVIF_LOCAL_ZLIBPNG` is set to `ON`. The at line 123 we have:

```
        set(ZLIB_LIBRARY zlibstatic)
```

I verified that at line 595, the value of `${ZLIB_LIBRARY}` is still `zlibstatic`. But the `target_link_libraries()` command at line 595 somehow magically substitutes `ext/zlib/libz.a` (rather than `zlibstatic`) for `${ZLIB_LIBRARY}`. I don't know how CMake does that.

Similarly for the `PNG_LIBRARY` and `JPEG_LIBRARY` variables. They are set to `png_static` and `jpeg` at line 119 and line 130 respectively, but when these two variables are used in the `target_link_libraries()` command at line 595, they are replaced by `ext/libpng/libpng16d.a` and `ext/libjpeg/libjpeg.a` magically.

Do you know how this works?

Note: I used the following commands to remove the zlib and jpeg system headers from my Linux desktop:
```
sudo apt remove zlib1g-dev:amd64
sudo apt remove libjpeg62-turbo-dev:amd64
```

Reply:
Thanks a lot for the explanation. It all makes sense now.

I still have a lot to learn about CMake...